### PR TITLE
Monsters will now attack. Additional checks have been added when generating spawn.

### PR DESCRIPTION
**Pull request by tumultenrx**
Monsters will now attack. Additional checks have been added when generating spawn.

Just want to help out, noticed monsters did not attack and sometimes spawn-point was generated in the middle of ocean or under ground.

### CODE REVIEW

File: src/World.cpp [hunk 1]
```diff
@@ -691,9 +696,56 @@ void cWorld::GenerateRandomSpawn(void)
...
-	LOGINFO("Generated random spawnpoint position {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+		LOGINFO("Generated random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	}
+	else
+	{
+		LOGINFO("Did not find an acceptable spawnpoint. Generated a random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	} // Maybe widen the search instead?
+
+}
+bool cWorld::CheckPlayerSpawnPoint(int posX, int posY, int posZ)
```
Comment (NiLSPACE) [inline src/World.cpp]: You need 5 empty lines before you start a new function.

File: src/World.h [hunk 2]
```diff
@@ -1077,6 +1077,9 @@ class cWorld :
 	/** <summary>Generates a random spawnpoint on solid land by walking chunks and finding their biomes</summary> */
 	void GenerateRandomSpawn(void);
 
+	/** Check if player starting point is acceptable **/
+	bool cWorld::CheckPlayerSpawnPoint(int posX, int posY, int posZ);
```
Comment (NiLSPACE) [inline src/World.h]: No need to have `cWorld::`  here

**Comment by tumultenrx** [comment]
Alright i will fix that thanks :)

File: src/World.h (hunk 2 above)
Comment (NiLSPACE) [inline src/World.h]: Also we start paramters with an `a_` prefix, and use PascalCase for the name. So `posX` would become `a_PosX` etc.

**Comment by tumultenrx** [comment]
Yeah noticed sorry!

**Comment by SphinxC0re** [comment]
I checked your code but I didn't see anything which checks whether the spawnpoint spawns the players in mid-air... so can players still spawn in mid-air?

**Comment by tumultenrx** [comment]
I have a check to make sure the block below is solid

**Comment by SphinxC0re** [comment]
Oh, I see :D

**Comment by SphinxC0re** [comment]
Also you should reconsider the naming of your commits

**Comment by tumultenrx** [comment]
Dont know how i name them :) Github noob

**Comment by SphinxC0re** [comment]
More like git noob :) which OS are you using?

**Comment by tumultenrx** [comment]
Windows using Git shell

File: src/World.cpp [hunk 3]
```diff
@@ -691,9 +696,61 @@ void cWorld::GenerateRandomSpawn(void)
...
+	if (foundSpawnPoint)
+	{
+		m_SpawnY = (double)GetHeight((int)m_SpawnX, (int)m_SpawnZ) + 1.6f;  // 1.6f to accomodate player height
+
+		LOGINFO("Generated random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	}
+	else
+	{
+		LOGINFO("Did not find an acceptable spawnpoint. Generated a random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	} // Maybe widen the search instead?
```
Comment (NiLSPACE) [inline src/World.cpp]: You need an extra space before starting the comment. That's why the build is failing. `}<space><space>// Mayb....`

**Comment by SphinxC0re** [comment]
https://progit.org/

**Comment by tumultenrx** [comment]
Thanks :)

File: src/Mobs/AggressiveMonster.cpp [hunk 4]
```diff
@@ -95,8 +95,7 @@ void cAggressiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 void cAggressiveMonster::Attack(std::chrono::milliseconds a_Dt)
 {
-	m_AttackInterval += a_Dt.count() * m_AttackRate;
-
+	m_AttackInterval += ((float)a_Dt.count() / 1000) * m_AttackRate;
```
Comment (madmaxoft) [inline src/Mobs/AggressiveMonster.cpp]: We prefer to use C++-style casts: `static_cast<float>(...)`

File: src/Mobs/Skeleton.cpp [hunk 5]
```diff
@@ -69,8 +69,7 @@ void cSkeleton::MoveToPosition(const Vector3d & a_Position)
 
 void cSkeleton::Attack(std::chrono::milliseconds a_Dt)
 {
-	m_AttackInterval += a_Dt.count() * m_AttackRate;
-
+	m_AttackInterval += ((float)a_Dt.count() / 1000) * m_AttackRate;
```
Comment (madmaxoft) [inline src/Mobs/Skeleton.cpp]: Again, use a C++-style cast

File: src/World.cpp [hunk 6]
```diff
@@ -691,9 +696,61 @@ void cWorld::GenerateRandomSpawn(void)
...
+			return false;
+		}
+	}  // for i - Coords[]
+	// Check if block below is solid
+	BLOCKTYPE BlockType = GetBlock(a_PosX, a_PosY - 1, a_PosZ);
+	if (!cBlockInfo::IsSolid(BlockType))
+	{
+		return false;
+	}
+	// Players should spawn with only air above.
```
Comment (madmaxoft) [inline src/World.cpp]: Add an empty line above each of these comments, and add a comment for the first code block.

**Comment by madmaxoft** [comment]
After you're done with the changes, it'd be great to squash all these commits into a single one. Use these commands:
```
git remote add upstream https://github.com/mc-server/MCServer.git
git fetch --all
git rebase --interactive upstream/master
  (a text editor opens with instructions on how to rebase, 
  follow those to squash all the commits into a single one)
git push --force
```
Note that you will have to force-push the results, that's because you're effectively editing history with these commands.

**Comment by tumultenrx** [comment]
Hopefully all is correct now :)

**Comment by tigerw** [comment]
Fairy Tail

\- Fairy Tail

**Pull request merged**